### DISCUSSION TITLE
Dependency between Geo and Query photo search terms

### DISCUSSION THREAD

**Discussion by kvalev** [photoprism/photoprism#2696] [Q&A] [answered]
I was looking into the search query which filters photos with geo information and stumbled upon the following block:

https://github.com/photoprism/photoprism/blob/94d2b3f7aee822d9a99aaddb31efef6d794ef9c6/internal/search/photos.go#L213-L221

I dont quite understand why there is a dependency between the `Geo` and the `Query` search terms. It seems the `Query` search term is only used when the user does not use the `Geo` one, which is not very intuitive. And the `k.keyword` query is somehow present in both blocks, although it has very little to do with the `Geo` one.

Shouldn't the handling of both search terms be decoupled? Is there some historical reason that I am missing?

**Comment by lastzero** [comment, marked as answer]
This might be for historical reasons, since we now have a separate endpoint that returns GeoJSON. What do you use the "geo:true" filter for? If it's important, I can put it on my to-do list when I get back to the office.

**Reply by kvalev** [reply]
I am trying to find all pictures that do not have gps information, so I am using "geo:true" to create a diff. On that note it would be nice if there was a way to directly query for all pictures _without_ geo information natively, instead of having to script that.
Its not super important, just nice to have

**Reply by lastzero** [reply]
Might add a GPS string filter for this, so you can use no or none as value. My guess is that we added geo for internal use, for example by the moments background worker.
